module/zfs/spa_stats.c

```diff
@@ -153,9 +152,7 @@ spa_read_history_init(spa_t *spa)
 	ssh->size = 0;
 	ssh->private = NULL;
 
-	(void) snprintf(name, KSTAT_STRLEN, "zfs/%s", spa_name(spa));
-
-	ksp = kstat_create(name, 0, "reads", "misc",
+	ksp = kstat_create("", 0, "reads", "misc",
```

Comment:
`kstat_create()` accepts the module name as the first parameter, why are we passing an empty string here just to set it manually later via `snprintf()`? Is this to minimize stack space usage?

What happens if `kstat_create()` does some kind of input validation on the first parameter and we are now bypassing it? Even if it's not doing it now this may change in the future.